Add DB model for EPG contracts

amittbose opened the pull request:
Have used an auto-increment column on the EPG table
to generate an internal ID that will be referenced
by the contracts table. This is needed for resources
that have non-scalar attributes (e.g. list of contracts).
Having an internal ID avoids having primary
keys that use many columns in the table that contains
the non-scalar values. Have updated other ACI model
classes with this internal-ID.

Signed-off-by: Amit Bose <amitbose@gmail.com>

ivar-lazzaro commented on aim/db/models.py:
```diff
@@ -40,57 +51,104 @@ class BridgeDomain(model_base.Base, model_base.HasName,
     bd_tenant_name = model_base.name_column()
-    # TODO(amitbose) Map contract names
+
+    contract = orm.relationship(EndpointGroupContract,
```
this should be contracts*

amittbose commented:
Addressed review comments:
* Renamed 'contract' to 'contracts' in EPG DB model
* Added indices for columns that have unique-constraint

For now I'm keeping the surrogate 'id' column as the primary key because I convinced myself by reading online discussions that doing so is a better approach :)
